### PR TITLE
Story 60 create button menu

### PR DESCRIPTION
**Pull request by ericdrosado**
screen shot 2017-08-17 at 1 56 49 pm

### CODE REVIEW

File: src/games/PlayerVsComputerGame.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+function PlayerVsComputerGame(ui, GameBoard, WinConditions) {
+  this.ui = ui;
+  this.GameBoard = GameBoard;
+  this.WinConditions = WinConditions;
+}
+
+PlayerVsComputerGame.prototype.initializeGame = function() {
+  this.playerMarker = "X";
+  this.ui.displayTurn(this.playerMarker);
+  console.log("In PvC");
```
Comment (tommcgee) [inline src/games/PlayerVsComputerGame.js]: We're not merging `console.log` anymore

File: src/index.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 var $ = require('jquery');
-var UI = require('./ui');
+// var UI = require('./ui');
```
Comment (tommcgee) [inline src/index.js]: or comments 👅

File: src/games/PlayerVsComputerGame.js [hunk 3]
```diff
@@ -7,7 +7,30 @@ function PlayerVsComputerGame(ui, GameBoard, WinConditions) {
...
+  var id = this.ui.spotClicked(e, this.playerMarker);
+  var gameBoard = this.GameBoard.updateBoard(id, this.playerMarker);
+  if (this.WinConditions.endGame(gameBoard)){
+    this.ui.displayWin(this.playerMarker);
+    this.ui.disableAllClicks();
+  } else if (this.GameBoard.isBoardFull()) {
+      this.ui.displayTie();
+  } else {
+      this.playerMarker = this.GameBoard.switchMarker(this.playerMarker);
+      PlayerVsComputerGame.prototype.simpleAI(this.ui, this.GameBoard, this.playerMarker);
```
Comment (tommcgee) [inline src/games/PlayerVsComputerGame.js]: This function could DEFINITELY use a better name! It looks like you're creating a class to me!

File: src/GameBoard.js [hunk 4]
```diff
@@ -22,4 +22,13 @@ GameBoard.prototype.switchMarker = function(playerMarker) {
   return playerMarker;
 }
 
+GameBoard.prototype.chooseAvailableSpace = function() {
+  for (i = 0; i <= 8; i++) {
+    var num = this.gameBoard[i].match(/[0-8]/);
```
Comment (tommcgee) [inline src/GameBoard.js]: can't this be simplified?

File: src/GameBoard.js [hunk 5]
```diff
@@ -22,4 +22,13 @@ GameBoard.prototype.switchMarker = function(playerMarker) {
...
+    if (this.gameBoard[i] == num ) {
```
Comment (tommcgee) [inline src/GameBoard.js]: `if (this.gameBoard[i] === i) {`

File: src/GameBoard.js (hunk 4 above)
Comment (tommcgee) [inline src/GameBoard.js]: Also not directly related to this PR, but very important (and something to think about before you dive to deep into minimax)... Does it make sense for your game board to look like ['0','1','2','3',4','5','6','7','8']? That seems like it _MIGHT_ make sense in a console version of the game (where the number is used to reference the spot you want to go in), but when you're using a UI like this there's no reason to map the index of each empty spot to the empty spot. Think about using something else to represent nothing being there!

File: src/games/PlayerVsComputerGame.js [hunk 6]
```diff
@@ -20,13 +20,13 @@ PlayerVsComputerGame.prototype.play = function(e) {
       this.ui.displayTie();
   } else {
       this.playerMarker = this.GameBoard.switchMarker(this.playerMarker);
-      PlayerVsComputerGame.prototype.simpleAI(this.ui, this.GameBoard, this.playerMarker);
+      PlayerVsComputerGame.prototype.pickRandomSpaceForAI(this.ui, this.GameBoard, this.playerMarker);
```
Comment (tommcgee) [inline src/games/PlayerVsComputerGame.js]: I think this should actually end up being it's own `Computer` Class that doesn't know about the UI  and simply just returns the index of the spot on the board it wants to go. Your game can handle actually placing the spot on the board!

File: src/TicTacToe.js [hunk 7]
```diff
@@ -26,18 +27,19 @@ TicTacToe.prototype.determineGameType = function(e) {
...
   this.GameBoard = new GameBoard();
   this.WinConditions = new WinConditions();
   this.game = new PlayerVsPlayerGame(this.ui, this.GameBoard, this.WinConditions);
   this.game.initializeGame();
 }
 
 TicTacToe.prototype.playerVsComputerGame = function() {
-  this.ui = new UI;
+  this.ComputerLogic = new ComputerLogic();
+  this.ui = new UI();
```
Comment (tommcgee) [inline src/TicTacToe.js]: you seem to define `this.ui` as the same thing in every method, any reason for this?

File: src/TicTacToe.js [hunk 8]
```diff
@@ -12,35 +12,22 @@ var TicTacToe = function() {
...
 }
 
-TicTacToe.prototype.determineGameType = function(e) {
-  this.ui = new UI();
-  var compiler = new HandlebarsCompiler();
-  var id = this.ui.menuButtonClicked(e);
+TicTacToe.prototype.determineGameType = function(e, ui) {
+  var game;
+  this.gameBoard = new GameBoard();
+  this.winConditions = new WinConditions();
```
Comment (NicoleCarpenter) [inline src/TicTacToe.js]: 🆒

File: src/TicTacToe.js [hunk 9]
```diff
@@ -12,35 +12,22 @@ var TicTacToe = function() {
 
 TicTacToe.prototype.menu = function() {
   this.ui = new UI();
-  this.ui.onButtonClicked(TicTacToe.prototype.determineGameType);
+  this.ui.onButtonClicked(TicTacToe.prototype.determineGameType, this.ui);
```
Comment (tommcgee) [inline src/TicTacToe.js]: A little weird, but after working with you on Friday this is probably the best solution 👍 Would be nice if we could figure something else out down the line!

Comment (tommcgee) [inline src/TicTacToe.js]: Thanks for the screenshot by the way! Very helpful.

**Pull request merged**